pkg/informer/informer.go

```diff
@@ -51,18 +51,66 @@ type clusterDiscovery interface {
 type DynamicDiscoverySharedInformerFactory struct {
 	workspaceLister tenancylisters.ClusterWorkspaceLister
 	disco           clusterDiscovery
-	dsif            dynamicinformer.DynamicSharedInformerFactory
+	dynamicClient   dynamic.Interface
 	handler         GVREventHandler
 	filterFunc      func(interface{}) bool
 	pollInterval    time.Duration
 
-	mu   sync.RWMutex // guards gvrs
-	gvrs sets.String
+	mu            sync.RWMutex // guards gvrs
+	gvrs          map[schema.GroupVersionResource]struct{}
+	informers     map[schema.GroupVersionResource]informers.GenericInformer
+	informerStops map[schema.GroupVersionResource]chan struct{}
 }
 
 // IndexerFor returns the indexer for the given type GVR.
 func (d *DynamicDiscoverySharedInformerFactory) IndexerFor(gvr schema.GroupVersionResource) cache.Indexer {
-	return d.dsif.ForResource(gvr).Informer().GetIndexer()
+	return d.InformerForResource(gvr).Informer().GetIndexer()
+}
+
+// InformerForResource returns the GenericInformer for gvr, creating it if needed.
+func (d *DynamicDiscoverySharedInformerFactory) InformerForResource(gvr schema.GroupVersionResource) informers.GenericInformer {
+	// See if we already have it
+	inf := func() informers.GenericInformer {
+		d.mu.RLock()
+		defer d.mu.RUnlock()
+		return d.informers[gvr]
+	}()
+
+	if inf != nil {
+		return inf
+	}
+
+	// Grab the write lock, then find-or-create
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	return d.informerForResourceLH(gvr)
```

Comment:
What does `LH` mean?

Reply:
LockHeld

Reply:
am used to WithLock suffix

Reply:
Just pushed LockHeld. That ok or want another rename?